src/Accelerators/NNPA/Conversion/ZHighToZLow/ZHighToZLow.cpp

```diff
@@ -1498,13 +1499,59 @@ struct ZHighToZLowBatchNormOpLowering : public ConversionPattern {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// Lower ZHigh StickifiedConstantOfShape to ZLow
+//===----------------------------------------------------------------------===//
+
+struct ZHighToZLowStickifiedConstantOfShapeOpLowering
+    : public ConversionPattern {
+  ZHighToZLowStickifiedConstantOfShapeOpLowering(
+      TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(typeConverter,
+            ZHighStickifiedConstantOfShapeOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op->getLoc();
+
+    auto stickOp = cast<ZHighStickifiedConstantOfShapeOp>(op);
+    FloatAttr value = stickOp.getValueAttr();
+
+    // Convert the scalar value to dlfloat16.
+    float valueF32 = (float)value.getValueAsDouble();
+    uint16_t valueDLF16; // Use uint16_t as container.
+    fp32_to_dlf16(&valueF32, &valueDLF16, 1);
+
+    MultiDialectBuilder<IndexExprBuilderForKrnl, MathBuilder, KrnlBuilder>
+        create(rewriter, loc);
+    ZHighStickifiedConstantOfShapeOpShapeHelper shapeHelper(
+        op, operands, &create.krnlIE);
+    shapeHelper.computeShapeAndAssertOnFailure();
+
+    // Convert ZTensor type to MemRefType.
+    ZMemRefType zMemRefType =
+        convertZTensorToMemRefType(*op->result_type_begin());
+
+    // Allocate a buffer for the result MemRef.
+    Value res = insertAllocAndDeallocZMemRef(
```

Comment:
Do we add deallocZMemRef buy ourselves? Or just it's an old name for the function?

Reply:
I removed `dealloc` in all names and comments. Thanks!